modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/tcs/TcsKeywordReader.scala

```diff
@@ -258,4 +271,16 @@ object TcsKeywordsReaderImpl extends TcsKeywordsReader {
   override def getOiwfsFreq: SeqAction[Option[Double]] = TcsEpics.instance.oiwfsIntegrationTime.map(calcFrequency)
 
   override def getCarouselMode: SeqAction[Option[String]] = TcsEpics.instance.carouselMode
+
+  override def getGnirsInstPort: SeqAction[Option[Int]] = TcsEpics.instance.gnirsPort
```

Comment:
Are the ports constants or do we read them from the A&G?

Reply:
They are read from the AG.